crates/matrix-sdk/src/media.rs

```diff
@@ -269,10 +272,19 @@ impl Media {
             }
         };
 
+        let server_versions = self.client.server_versions().await?;
```

Comment:
This allows the homeserver to downgrade us towards unauthenticated media uploads. While it isn't the end of the world, I think we should provide a `ClientBuilder` option that allows users to always use the authenticated endpoints, despite what the homeserver suggests.

This doesn't need to happen as part of this PR, but please leave a TODO item here.

Reply:
Todo added